dodola/repository.py

```diff
@@ -2,142 +2,58 @@
-        return x
-
-    def write(self, url_or_path, x):
-        """Write Dataset to Zarr store
-
-        This opens Zarr store with mode "w" and is called with with
-        ``compute=True``, so any lazy computations will be completed.
 
-        Parameters
-        ----------
-        url_or_path : str
-            Location to write Zarr store to.
-        x : xr.Dataset
-        """
-        x.to_zarr(self.get_mapper(url_or_path), mode="w", compute=True)
-        logger.info(f"Written {url_or_path}")
+def write(url_or_path, x):
+    """Write Dataset to Zarr store
 
-    def get_mapper(self, root, check=False, create=False):
-        """Get fsspec.FSMap from wrapped FileSystemAbstraction
-
-        Parameters
-        ----------
-        root : str
-        check : bool
-            Do touch at storage to check for write access.
-        create : bool
-
-        Returns
-        -------
-        out : fsspec.FSMap
-        """
-        return self._fs.get_mapper(root, check=check, create=create)
-
-
-def adl_repository(
-    account_name=None,
-    account_key=None,
-    client_id=None,
-    client_secret=None,
-    tenant_id=None,
-):
-    """Instantiate a Zarr data repo using Azure Datalake Gen. 2 as backend
-
-    To authenticate with storage must pass in `account_name` and
-    `account_key` for key-based authentication or `client_id`, `client_secret`,
-    and `tenant_id` for authentication with service principal
-    credentials. Initializing arguments are passed to ``adlfs.AzureBlobFileSystem``.
+    This opens Zarr store with mode "w" and is called with with
+    ``compute=True``, so any lazy computations will be completed.
 
     Parameters
     ----------
-    account_name : str or None, optional
-    account_key : str or None, optional
-    client_id : str or None, optional
-    client_secret : str or None, optional
-    tenant_id : str or None, optional
-
-    Returns
-    -------
-    repo : dodola.repository._ZarrRepo
+    url_or_path : str
+        Location to write Zarr store to.
+    x : xr.Dataset
     """
-    repo = _ZarrRepo(
-        fs=AzureBlobFileSystem(
-            account_name=account_name,
-            account_key=account_key,
-            client_id=client_id,
-            client_secret=client_secret,
-            tenant_id=tenant_id,
-        )
-    )
-    return repo
-
+    x.to_zarr(url_or_path, mode="w", compute=True)
+    logger.info(f"Written {url_or_path}")
 
-def memory_repository(storage=None):
-    """Instantiate a Zarr data repo in memory
 
-    This is most commonly used for testing. Keep in mind that all instances
-    share memory globally.
+def get_mapper(root, check=False, create=False):
+    """Get fsspec.FSMap from wrapped FileSystemAbstraction
 
     Parameters
     ----------
-    storage : dict or None, optional
-        Mapping of `{path: data}` to immediately store into memory filesystem.
+    root : str
+    check : bool
+        Do touch at storage to check for write access.
```

Comment:
I'm a little unclear on what the description of `check` means here...

Reply:
This was removed all together in PR #60. It's no longer an issue. Marking as resolved.